src/compiler/program.ts

```diff
@@ -1521,6 +1758,57 @@ namespace ts {
             });
         }
 
+        function processTypeReferenceDirectives(file: SourceFile, compilationRoot: string) {
+            const typeDirectives = map(file.typeReferenceDirectives, l => l.fileName);
+            const resolutions = resolveTypeReferenceDirectiveNamesWorker(typeDirectives, file.fileName);
+
+            for (let i = 0; i < typeDirectives.length; i++) {
+                const ref = file.typeReferenceDirectives[i];
+                const resolvedTypeReferenceDirective = resolutions[i];
+                // store resolved type directive on the file
+                setResolvedTypeReferenceDirective(file, ref.fileName, resolvedTypeReferenceDirective);
+                // If we already found this library as a primary reference, or failed to find it, nothing to do
+                const previousResolution = resolvedTypeReferenceDirectives[ref.fileName];
+                if (previousResolution && (previousResolution.primary || (previousResolution.resolvedFileName === undefined))) {

```

Comment:
do not think you need to check for previousResolution.resolvedFileName === undefined